From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yocto Build <build@yoctoproject.org>
Date: Thu, 12 Dec 2024 00:00:00 +0000
Subject: [PATCH] Use PkgConfig to find libconfig for Yocto builds

Yocto's libconfig package doesn't provide CMake config files, so use
pkg-config instead to locate the library.

Upstream-Status: Inappropriate [Yocto-specific]
Signed-off-by: Yocto Build <build@yoctoproject.org>
---
 src/native/tools/recompress/CMakeLists.txt | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/src/native/tools/recompress/CMakeLists.txt b/src/native/tools/recompress/CMakeLists.txt
index 8688bb7..69be641 100644
--- a/src/native/tools/recompress/CMakeLists.txt
+++ b/src/native/tools/recompress/CMakeLists.txt
@@ -4,14 +4,12 @@ add_executable(recompress
 	swupdate_helpers.cpp
 	)
 
-find_package(libconfig CONFIG REQUIRED)
+# Use PkgConfig to find libconfig since it doesn't provide CMake config in Yocto
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LIBCONFIG REQUIRED libconfig++)
 
-if(UNIX)
-	target_link_libraries(recompress PRIVATE stdc++fs)
-	target_link_libraries(recompress PRIVATE libconfig::config libconfig::config++)
-else()
-	target_link_libraries(recompress PRIVATE libconfig::libconfig libconfig::libconfig++)
-endif()
+target_link_libraries(recompress PRIVATE stdc++fs)
+target_link_libraries(recompress PRIVATE ${LIBCONFIG_LIBRARIES})
 
 target_link_libraries(recompress
 	PRIVATE
-- 
2.39.0
